keploy/keploy.go

```diff
@@ -750,54 +965,78 @@ func (k *Keploy) newGet(url string) ([]byte, error) {
 		return nil, err
 	}
 	return body, nil
+
 }
 
 // fetch makes a get request to keploy API server and returns array of testcases
 func (k *Keploy) fetch() []models.TestCase {
 
 	var tcs []models.TestCase = []models.TestCase{}
 	pageSize := 25
-
 	for i := 0; ; i += pageSize {
-		url := fmt.Sprintf("%s/regression/testcase?app=%s&offset=%d&limit=%d&testCasePath=%s&mockPath=%s", k.cfg.Server.URL, k.cfg.App.Name, i, 25, k.cfg.App.TestPath, k.cfg.App.MockPath)
+		var res []models.TestCase
+		if k.cfg.Server.GrpcEnabled {
+			resp, err := k.grpcClient.GetTCS(k.Ctx, &proto.GetTCSRequest{
+				App:          k.cfg.App.Name,
+				Offset:       strconv.Itoa(i),
+				Limit:        strconv.Itoa(pageSize),
+				TestCasePath: k.cfg.App.TestPath,
+				MockPath:     k.cfg.App.MockPath,
+			})
+			if err != nil {
+				k.Log.Error("failed to fetch testcases from keploy cloud", zap.Error(err))
+				return nil
+			}
+			res = ProtoToModelsTestCase(resp.GetTcs())
+			tcs = append(tcs, res...)
+			if len(res) < pageSize {
+				break
+			}
+			if resp.Eof {
+				break
+			}
 
-		req, err := http.NewRequest("GET", url, http.NoBody)
-		if err != nil {
-			k.Log.Error("failed to fetch testcases from keploy cloud", zap.Error(err))
-			return nil
-		}
-		k.setKey(req)
-		resp, err := k.client.Do(req)
-		if err != nil {
-			k.Log.Error("failed to fetch testcases from keploy cloud", zap.Error(err))
-			return nil
-		}
-		if resp.StatusCode != http.StatusOK {
-			k.Log.Error("failed to fetch testcases from keploy cloud", zap.Error(errors.New("failed to send get request: "+resp.Status)))
-			return nil
-		}
+		} else {
+			url := fmt.Sprintf("%s/regression/testcase?app=%s&offset=%d&limit=%d&testCasePath=%s&mockPath=%s&reqType=%s", k.cfg.Server.URL, k.cfg.App.Name, i, 25, k.cfg.App.TestPath, k.cfg.App.MockPath)
```

Comment:
Please remove the reqType query parameter.

Reply:
Done. PTAL